src/acl.c

```diff
@@ -2738,38 +2736,40 @@ void aclCommand(client *c) {
     {
         addReplyError(c,"This Redis instance is not configured to use an ACL file. You may want to specify users via the ACL SETUSER command and then issue a CONFIG REWRITE (assuming you have a Redis configuration file set) in order to store users in the Redis configuration.");
         return;
-    } else if (!strcasecmp(sub,"load") && c->argc == 2) {
+    } else if (!strcasecmp(sub,"load")) {
         sds errors = ACLLoadFromFile(server.acl_filename);
         if (errors == NULL) {
             addReply(c,shared.ok);
         } else {
             addReplyError(c,errors);
             sdsfree(errors);
         }
-    } else if (!strcasecmp(sub,"save") && c->argc == 2) {
+    } else if (!strcasecmp(sub,"save")) {
         if (ACLSaveToFile(server.acl_filename) == C_OK) {
             addReply(c,shared.ok);
         } else {
             addReplyError(c,"There was an error trying to save the ACLs. "
                             "Please check the server logs for more "
                             "information");
+        }        
+    } else if (!strcasecmp(sub,"cat") && (c->argc == 2 || c->argc == 3)) {
+        if (c->argc == 2) {
```

Comment:
I think this change to nested if statements affects the git log unnecessarily. Checking `strcasecmp` against "cat" twice here is hardly bottleneck.

Reply:
I agree with Viktor. but also i wonder if we really need that entire PR.
It's true that now we have arity checks for sub commands and these are not needed, but maybe we can view them as a safety thing (in case there's a mismatch).
i wouldn't have bothered to add them if they were missing, but also not sure we want to change so many lines to remove something that's not harmful.

Reply:
TBH i just wanted to get rid of some code in `xinfoCommand`, maybe i took it too far
i'm going to close this PR and open two others:
1. for the small cleanup in xinfoCommand
2. for the weird behavior of MEMORY USGAE (seems like it can take the work SAMPLES multiple times for no apparent reason)
```
127.0.0.1:6379> MEMORY USAGE x1 samples 4
(integer) 1772
127.0.0.1:6379> MEMORY USAGE x1 samples 4 samples 7
(integer) 1772
```

Reply:
regarding repeated arguments, note that it may be a breaking change.
see https://github.com/redis/redis/pull/10419